### PR TITLE
use pre-commit for python files

### PR DESCRIPTION
**Pull request by 3nids**
This enables pre-commit hook for Python files and disable Python formatting from prepare-commit.sh

I've only added: pyupgrade, black and isort without specific config for the sake of simplicity/clarity. See config for details.

I've also enabled auto-fix for PRs (will be effective when we activate pre-commit.ci).

We've recently succeeded at using clang-format on a sipified project (see https://github.com/kadas-albireo/kadas-albireo2/pull/175). So this is a first small step before starting to touch at cpp code.

### CODE REVIEW

**Comment by 3nids** [comment]
@troopa81 @nyalldawson @elpaso @m-kuhn any feedback welcome!

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  - repo: https://github.com/pycqa/isort
+    rev: 5.13.2
+    hooks:
+      - id: isort
+        args: ["--profile", "black", "--filter-files"]
+
+ci:
+  autofix_prs: true
+  autoupdate_schedule: quarterly
```
Comment (troopa81) [inline .pre-commit-config.yaml:27]: Would that mean that if you're the first to push a PR after a hook update (let's say black rules for instance), you're gonna push a lot of modified files which are not related to your PR? Or is there an auto PR/commit pushed automatically on the repo on update?

**Comment by m-kuhn** [comment]
I am a big fan of `pre-commit` and supportive of actions to move there (I even tried to launch an unsuccessful [crowdfunding campaign ](https://www.linkedin.com/posts/kuhnmatthias_qgis-opensourcecommunity-geospatial-activity-7194924382308233216-116I) to sponsor @elpaso 's [QEP](https://github.com/qgis/QGIS-Enhancement-Proposals/issues/265)). I also like `black`. I did not review the changes or configuration yet.

The main question here for me is, how this coexists with existing pre-commit hooks, in terms of local development configuration? Does one call the other (`pre-commit` the existing `prepare_commit.sh` or vice versa) or can the system be configured to run both in a different way)?

File: .pre-commit-config.yaml (hunk 1 above)
Comment (3nids) [reply .pre-commit-config.yaml:27]: autoupdate is to update the tools used, so nothing related to the source code.

if there's a hook update, there's not that many changes brought. And in a PR, IIRC only the modified files are tested.

**Comment by nyalldawson** [comment]
Can you create a .git-blame-ignore-revs file and exclude the "run-precommit" commit from it?  See  https://gist.github.com/kateinoigakukun/b0bc920e587851bfffa98b9e279175f2

And I'm hugely in favour of this move!

**Comment by 3nids** [comment]
>  how this coexists with existing pre-commit hooks

good point, I've added prepare_commit to the pre-commit hook: [f24de2e](https://github.com/qgis/QGIS/pull/59522/commits/f24de2ebe6c62922174626afa75903b049ffeabf)

> Can you create a .git-blame-ignore-revs file and exclude the "run-precommit" commit from it? 

good catch. I've added the file. I suppose we can wait to merge to update that file with the proper commit (in case it's rebased)?

**Comment by 3nids** [comment]
@m-kuhn I realize now that doing this double config in pre-commit hook will prevent us from using pre-commit.ci service because there is no way to control additional dependencies for local/system repo. We could add a pip install on prior to the prepare_commit call but it looks not clean and might slow down the process. So we'll need to keep our own workflows until we can completely drop prepare_commit.sh.

**Comment by 3nids** [comment]
In case there is no further discussions, I would merge it in a couple of days.

I would assume the general acceptance is fine since there were no objections in @elpaso's QEP https://github.com/qgis/QGIS-Enhancement-Proposals/issues/265

**Review by m-kuhn — APPROVED** [review]
No objections so far but positive feedback. I'm in favor of merging this.
